src/test/java/com/meilisearch/integration/TenantTokenTest.java

```diff
@@ -0,0 +1,244 @@
+package com.meilisearch.integration;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import com.meilisearch.integration.classes.AbstractIT;
+import com.meilisearch.integration.classes.TestData;
+import com.meilisearch.sdk.Client;
+import com.meilisearch.sdk.Config;
+import com.meilisearch.sdk.Index;
+import com.meilisearch.sdk.Key;
+import com.meilisearch.sdk.Settings;
+import com.meilisearch.sdk.Task;
+import com.meilisearch.sdk.TenantTokenOptions;
+import com.meilisearch.sdk.exceptions.MeiliSearchException;
+import com.meilisearch.sdk.model.SearchResult;
+import com.meilisearch.sdk.utils.Movie;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+@Tag("token")
+public class TenantTokenTest extends AbstractIT {
+
+    private TestData<Movie> testData;
+
+    @BeforeEach
+    public void initialize() {
+        this.setUp();
+    }
+
+    @AfterAll
+    static void cleanMeiliSearch() {
+        cleanup();
+    }
+
+    /** Test Create Tenant Token with only search rules */
+    @Test
+    public void testGenerateTenantTokenWithOnlySearchRules() throws Exception {
+        String indexUid = "BasicGenerateToken";
+        Index index = createEmptyIndex(indexUid);
+        Key key = getPrivateKey();
+
+        Client privateClient = new Client(new Config("http://localhost:7700", key.getKey()));
+
+        Map<String, Object> rules = new HashMap<String, Object>();
+        rules.put("*", new HashMap<String, Object>());
+
+        String jwtToken = privateClient.generateTenantToken(rules);
+
+        Client tokenClient = new Client(new Config("http://localhost:7700", jwtToken));
+
+        SearchResult searchResult = tokenClient.index(indexUid).search("");
```

Comment:
Since you threw an `Exception` if something goes wrong in the search request you can just ensure there is no Exception after the request :)